actor_library/src/token/erc20.rs

```diff
@@ -0,0 +1,560 @@
+    ) -> anyhow::Result<()>;
+
+    fn allowance<BS: Blockstore>(&self, store: &BS, owner: Address, spender: Address) -> BigUint;
+
+    fn approve<BS: Blockstore>(
+        &mut self,
+        store: &BS,
+        spender: Address,
+        amount: BigUint,
+    ) -> anyhow::Result<()>;
+
+    fn transfer_from<BS: Blockstore>(
+        &mut self,
+        store: &BS,
+        from: Address,
+        to: Address,
+        amount: BigUint,
+    ) -> anyhow::Result<()>;
+}
+
+pub trait ERC20Metadata {
+    fn name(&self) -> String;
+
+    fn symbol(&self) -> String;
+
+    fn decimals(&self) -> u8;
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct DefaultERC20 {
+    //Mapping(Address -> u256)
+    balances: Cid,
+    //Mapping(owner_address + spender_address -> u256)
+    allowances: Cid,
+    total_supply: BigUint,
+    name: String,
+    symbol: String,
+}
+
+impl Context for DefaultERC20 {}
+
+impl DefaultERC20 {
+    pub fn new<BS: Blockstore>(
+        store: &BS,
+        name: String,
+        symbol: String,
+    ) -> anyhow::Result<DefaultERC20> {
+        let empty_balances_map = map::make_empty_map::<_, BigUint>(store).flush()?;
+        let empty_allowances_map = map::make_empty_map::<_, BigUint>(store).flush()?;
+
+        Ok(Self {
+            balances: empty_balances_map,
+            allowances: empty_allowances_map,
+            total_supply: BigUint::default(),
+            name,
+            symbol,
+        })
+    }
+
+    pub fn increase_allowance<BS: Blockstore>(
+        &mut self,
+        store: &BS,
+        spender: Address,
+        added_value: BigUint,
+    ) -> anyhow::Result<()> {
+        let owner = DefaultERC20::msg_sender();
+        let spender = match resolve_address(&spender) {
+            Some(spender) => spender,
+            None => {
+                return Err(anyhow!(
+                    "ERC20: cannot resolve address in initial balance table"
+                ))
+            }
+        };
+
+        self.approve_tokens(
+            store,
+            owner,
+            spender,
+            self.get_allowance(store, owner, spender) + added_value,
+        )
+    }
+
+    pub fn decrease_allowance<BS: Blockstore>(
+        &mut self,
+        store: &BS,
+        spender: Address,
+        subtracted_value: BigUint,
+    ) -> anyhow::Result<()> {
+        let owner = DefaultERC20::msg_sender();
+
+        let spender = match resolve_address(&spender) {
+            Some(spender) => spender,
+            None => {
+                return Err(anyhow!(
+                    "ERC20: cannot resolve address in initial balance table"
+                ))
+            }
+        };
+
+        let current_allowance = self.get_allowance(store, owner, spender);
+
+        assert!(
+            current_allowance >= subtracted_value,
+            "ERC20: decreased allowance below zero"
+        );
+        self.approve_tokens(store, owner, spender, current_allowance - subtracted_value)
```

Comment:
Fix spacing so it matches with one in increase_allowance
```suggestion
        self.approve_tokens(
            store, 
            owner, 
            spender, 
            current_allowance - subtracted_value
        )
```

Reply:
This formatting is result of "cargo fmt --all" command.